File: app/src/main/java/com/rosberry/android/jigit/presentation/launch/LaunchPresenter.kt
/*
 *
 *  * Copyright (c) 2018 Rosberry. All rights reserved.
 *
 */

package com.rosberry.android.jigit.presentation.launch

import com.arellomobile.mvp.InjectViewState
import com.arellomobile.mvp.MvpPresenter
import com.arellomobile.mvp.MvpView
import com.rosberry.android.jigit.Screens
import com.rosberry.mpp.jigitbl.domain.AuthInteractor
import ru.terrakok.cicerone.Router
import javax.inject.Inject

/**
 * @author Alexei Korshun on 06/02/2019.
 */
@InjectViewState
class LaunchPresenter @Inject constructor(
        private val authInteractor: AuthInteractor,
        private val router: Router
) : MvpPresenter<MvpView>() {

    override fun onFirstViewAttach() {
        super.onFirstViewAttach()
        if (authInteractor.isAuth().not()) router.newRootScreen(Screens.LoginScreen())
        else router.newRootScreen(Screens.RepositoriesScreen())
    }
}

File: iosApp/jiGit/jiGit/src/presentation/launch/LaunchPresenter.swift
//
//  LaunchPresenter.swift
//  jiGit
//
//  Created by Aleksei Korshun on 08/02/2019.
//  Copyright © 2019 Aleksei Korshun. All rights reserved.
//

import FRAMEWORK

class LaunchPresenter {
 
    private let authInteractor: AuthInteractor
    private let router: LaunchRouter
    
    init(_ authInteractor: AuthInteractor, _ router: LaunchRouter) {
        self.authInteractor = authInteractor
        self.router = router
    }
    
    func onAttach() {
        if (!authInteractor.isAuth()) {
            router.navigateToLoginScreen()
        } else {
            router.navigateToRepositoriesScreen()
        }
    }
}

protocol LaunchRouter {
    
    func navigateToLoginScreen()

    func navigateToRepositoriesScreen()
}
